compaction/compaction_manager.cc

```diff
@@ -291,19 +291,19 @@ compaction_task_executor::compaction_task_executor(compaction_manager& mgr, tabl
     , _compacting_table(t)
     , _compaction_state(_cm.get_compaction_state(t))
     , _type(type)
-    , _gate_holder(_compaction_state.gate.hold())
     , _description(std::move(desc))
 {}
 
 future<compaction_manager::compaction_stats_opt> compaction_manager::perform_task(shared_ptr<compaction_task_executor> task) {
     _tasks.push_back(task);
+    gate::holder _gate_holder = task->_compaction_state.gate.hold();
```

Comment:
this should be done before `push_back(task)` since if the gate is closed, `task` will remain in `_tasks` without the `unregister_task` deferred action below to remove it.

Reply:
also, no need for the leading underscore in the variable name, as it's not a private member here.

Reply:
LGTM otherwise